### PR TITLE
feat: update x/foundation to use Finschia/cosmos-sdk

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

closes: #1197

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Comment by jaeseung-bae** [comment]
I think I deleted bump50 by accident 2 weeks ago.
I didn't know the deletion. Sorry for the inconvenience.

**Comment by 0Tech** [comment]
Blocked by #1205.

File: x/foundation/codec/cdc.go
Comment (ulbqb) [inline x/foundation/codec/cdc.go:1]: https://github.com/cosmos/cosmos-sdk/pull/16062
This file isn't needed. Amino codec is deprecated.

**Comment by ulbqb** [comment]
It's ok not to update simapp/app.go?

File: x/foundation/module/module.go
Comment (ulbqb) [inline x/foundation/module/module.go:220]: Please set a foundation keeper as struct. Stakingplus module can't use foundation keeper because foundation keeper struct is not register to simapp. Some tests of simapp has following error.

```
--- FAIL: TestInitCmd (0.15s)
panic: can't resolve type github.com/Finschia/finschia-sdk/x/stakingplus/stakingplus.FoundationKeeper for github.com/Finschia/finschia-sdk/x/stakingplus/module.ProvideModule
```

Comment (0Tech) [reply x/foundation/module/module.go:220]: I've updated the keeper.

**Comment by ulbqb** [comment]
How about removing simapp/app if we don't use app_v1?

File: x/foundation/README.md
Comment (ulbqb) [inline x/foundation/README.md:6]:
```suggestion
---
sidebar_position: 1
---
```

File: x/foundation/README.md
Comment (ulbqb) [inline x/foundation/README.md:87]: Please update all old links.

Comment (0Tech) [reply x/foundation/README.md:87]: I will update the links after this PR merged, because it would be squashed.

**Comment by 0Tech** [comment]
> How about removing simapp/app if we don't use app_v1?

I agree. Any opinions @jaeseung-bae @tkxkd0159 ?

**Comment by jaeseung-bae** [comment]
> > How about removing simapp/app if we don't use app_v1?
> 
> I agree. Any opinions @jaeseung-bae @tkxkd0159 ?

After we remove it, no further effort on app_v1 is expected when we track upstream's code, isn't it??

**Review by ulbqb — APPROVED** [review]
LGTM

**Review by jaeseung-bae — APPROVED** [review]
LGTM

**Review by ulbqb — APPROVED** [review]

**Pull request merged**
